### PR TITLE
Add a contrib document to show how to test scaffolder templates locally

### PR DESCRIPTION
**Pull request by Pike**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by Pike** [comment]
This is an idea on how to test scaffolder templates, building upon the dry-run scaffolder API.

@Rugvip , is this a good idea or bad abuse of that API?

Also, interested in what other folks think that talk about testing templates often.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice idea, thanks! 👍 

I'm thinking we would eventually want to build this into the scaffolder somehow if this has a lot of usage, since it's a bit tricky to use from contrib. What I'm thinking we could do is open up a feedback issue, link to that issue from the README here in contrib. Then we ask people to chime in on the issue if they have any feedback, good or bad, or even just saying that they're using it. What do you think?

**Comment by Pike** [comment]
ping @Rugvip

File: contrib/scaffolder/template-testing-dry-run.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  await writeResultContents(target, resultContents);
+};
+
+const main = async (argv: string[]) => {
+  program
+    .name('backstage-scaffolder')
+    .version(version)
+    .description('Creates a dry-run output of a given template')
+    .option('-t, --token <value>', 'JWT to use for auth')
+    .option('-b, --backend-secret <value>', 'Base64 encoded backend secret')
```
Comment (Rugvip) [inline contrib/scaffolder/template-testing-dry-run.md:151]: Not sure what happened here, could've sworn I left a comment somewhere about this 😅 

I don't want to encourage usage of backend secrets outside of the backend in any way to be honest. This is not a secret that you should be throwing around on dev machines in any way. In an optimal setup, nobody even touched or touches the secret during its lifecycle. Let's skip this option and just go with the token approach for where it's needed. I do want to get to a point where we can have a nicer flow, but we're not there yet.

**Review by Rugvip — CHANGES_REQUESTED** [review]

File: contrib/scaffolder/template-testing-dry-run.md (hunk 1 above)
Comment (Pike) [reply contrib/scaffolder/template-testing-dry-run.md]: Adjusted. We might back out the back-out internally to do CI, but I see why you wouldn't want to recommend that.

I've also added some `ora` use for user feedback.

Comment (Rugvip) [reply contrib/scaffolder/template-testing-dry-run.md]: Yep that's all good! And not gonna try to hide the history or anything, just wanna be careful with our recommendations

**Review by Rugvip — APPROVED** [review]
Nice, thank you!  👍

Let's :shipit:

**Pull request merged**